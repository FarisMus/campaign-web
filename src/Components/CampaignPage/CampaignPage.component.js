import React, { useState, useEffect } from 'react';
import useAsyncEffect from 'use-async-effect';
import { useSelector } from 'react-redux';

import { getCampaignList } from '../../Services/CampaignServices';
import CampaignCard from '../CampaignCard/CampaignCard.component';
import { sortObjectByKey } from '../../Utils/Utils'
import styles from './CampaignPage.componens.styles';

const CampaignPage = () => {
	const [campaigns, setCampaigns] = useState([]);
	const sortCriteria = useSelector((state) =>  state.sortBy);

	useAsyncEffect(async () => {
		const result = await getCampaignList();

		setCampaigns(result, sortCriteria);
	}, []);

	useEffect(() => {
		const sortedList = sortObjectByKey(campaigns, sortCriteria);

		setCampaigns(sortedList);
	}, [sortCriteria]);

	return (
		<div style={styles.container}>
			{campaigns.map((campaign) => <CampaignCard {...campaign} />)}
		</div>
	);
};

export default CampaignPage;
