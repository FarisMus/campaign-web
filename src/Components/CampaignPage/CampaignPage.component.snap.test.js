import React from 'react';
import renderer from 'react-test-renderer';
import CampaignPage from './CampaignPage.component';

jest
	.mock('react-redux', () => ({
	useDispatch: jest.fn(),
	useSelector: jest.fn()
}));

test('Render Campaign Page Component', () => {
	const component = renderer.create(
		<CampaignPage />,
	);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
