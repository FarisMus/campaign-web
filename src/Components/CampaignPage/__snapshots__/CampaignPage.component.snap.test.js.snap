// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Campaign Page Component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
/>
`;
